src/eureka/S5_lightcurve_fitting/models/FleckModel.py

```diff
@@ -0,0 +1,245 @@
+                        # Update the non-dictionary form as well
+                        setattr(self.parameters, item,
+                                self.parameters.dict[item])                     
+
+    def eval(self, channel=None, pid=None, **kwargs):
+        """Evaluate the function with the given values.
+
+        Parameters
+        ----------
+        channel : int; optional
+            If not None, only consider one of the channels. Defaults to None.
+        pid : int; optional
+            Planet ID, default is None which combines the eclipse models from
+            all planets.
+        **kwargs : dict
+            Must pass in the time array here if not already set.
+
+        Returns
+        -------
+        lcfinal : ndarray
+            The value of the model at the times self.time.
+        """
+        if channel is None:
+            nchan = self.nchannel_fitted
+            channels = self.fitted_channels
+        else:
+            nchan = 1
+            channels = [channel, ]
+            
+        if pid is None or pid == 0:
+            pid_iter = range(self.num_planets)
+            pid_id = ''
+        else:
+            pid_iter = [pid,]
+            pid_id = f'_pl{pid}'
+
+        # Get the time
+        if self.time is None:
+            self.time = kwargs.get('time')
+
+        # find number of spots
+        spotinds = [s for s in self.parameters.dict.keys() if 'spotrad' in s]   
+        counter = [s for s in spotinds if '_' in s]
+        nspots = len(spotinds) - len(counter)
+        
+        # Get relevant spot parameters
+        spotrad = np.zeros((self.nchannel_fitted, nspots))
+        spotlat = np.zeros((self.nchannel_fitted, nspots))
+        spotlon = np.zeros((self.nchannel_fitted, nspots))
+        spotcon = np.zeros((self.nchannel_fitted))
+        starrot = np.zeros((self.nchannel_fitted))
+        starinc = np.zeros((self.nchannel_fitted))
+
+        # Set all parameters
+        lcfinal = np.array([])
+        for c in range(nchan):
+            if self.nchannel_fitted > 1:
+                chan = channels[c]
+            else:
+                chan = 0
+
+            # channel ID
+            if chan == 0:
+                channel_id = ''
+            else:
+                channel_id = f'_{chan}'
+
+            time = self.time
+            if self.multwhite:
+                # Split the arrays that have lengths of the original time axis
+                time = split([time, ], self.nints, chan)[0]
+                
+            light_curve = np.ma.ones(time.shape)
+            for pid in pid_iter:
+                # Initialize planet
+                pl_params = PlanetParams(self, pid, chan)
+
+                # Set spot/fleck parameters
+                # set a star rotation for the star object
+                # not actually used in fast mode. 
+                # overwritten if user supplies and runs slow mode
+                starrot[:] = 100
```

Comment:
Can just do `starrot = 100*np.ones((lc.nchannel_fitted))` above instead of doing this over multiple lines